File: api/Core/Model/ProductType.cs
using System.ComponentModel.DataAnnotations;

namespace Xcianify.Core.Model
{
    public class ProductType : BaseModel
    {
        [Required]
        [StringLength(10)]
        public string Code { get; set; } = string.Empty;

        [Required]
        [StringLength(100)]
        public string Name { get; set; } = string.Empty;

        public string? Description { get; set; }

        public int? ParentTypeId { get; set; }

        public bool IsActive { get; set; } = true;

        // Navigation properties (for Dapper mapping)
        public ProductType? ParentType { get; set; }
        public List<ProductType> ChildTypes { get; set; } = new List<ProductType>();
    }
} 

File: pharma-erp-3/types/product-type.ts
export interface ProductType {
  id: number;
  code: string;
  name: string;
  description?: string;
  parentTypeId?: number;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
  createdBy?: number;
  updatedBy?: number;
  parentType?: ProductType;
  childTypes?: ProductType[];
}

export interface CreateProductTypeData {
  code: string;
  name: string;
  description?: string;
  parentTypeId?: number;
  isActive: boolean;
}

export interface UpdateProductTypeData {
  code: string;
  name: string;
  description?: string;
  parentTypeId?: number;
  isActive: boolean;
} 